### ISSUE TITLE
The default SQL generated for DATEADD() is an empty string

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11266]
Translating `dateadd(day, a, b)` to the `SQLDialect.DEFAULT` dialect produces an empty string, which in turn, can produce confusing syntax errors when using it in SQL like this:

```sql
-- Source
select 1, dateadd(day, a, b)

-- Output
select 1,
```

----
See also: https://github.com/jOOQ/jOOQ/issues/9085

**Comment by lukaseder** [comment]
The `FIREBIRD` generated SQL seems to be the best choice here.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11267)
